### PR TITLE
Add automatic and manual timezone support

### PR DESCRIPTION
**Pull request by miguelespinoza**
description from MM master

#### Summary
Add timezone support for mobile. Following these 2 PRs:
https://github.com/mattermost/mattermost-webapp/pull/696
https://github.com/mattermost/mattermost-server/pull/8185

#### Device Information
This PR was tested on: [iPhone 6 simulator, Nexus 5 emulator] 

##### iOS
ios_timezone

##### Android

android_timezone

### In action

timezone change behavior

### CODE REVIEW

File: app/screens/timezone/select_timezone/select_timezone.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        const {theme, initialScrollIndex} = this.props;
+        const {value} = this.state;
+        const {intl} = this.context;
+        const style = getStyleSheet(theme);
+
+        const searchBarInput = {
+            backgroundColor: changeOpacity(theme.sidebarHeaderTextColor, 0.2),
+            color: theme.sidebarHeaderTextColor,
+            fontSize: 15,
+            lineHeight: 66
```
Comment (jarredwitt) [inline app/screens/timezone/select_timezone/select_timezone.js]: Might check to see if this is needed. I had to remove this to get text to show correctly on Android.

File: app/screens/timezone/timezone.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        };
+
+        const updatedUser = {
+            ...user,
+            timezone
+        };
+
+        const {error} = await this.props.actions.updateUser(updatedUser);
+        if (error) {
+            this.handleRequestError(error);
```
Comment (jarredwitt) [inline app/screens/timezone/timezone.js]: Does this exist?

**Review by jarredwitt — CHANGES_REQUESTED** [review]

File: app/mattermost.js [hunk 3]
```diff
@@ -55,13 +56,16 @@ import {deleteFileCache} from 'app/utils/file';
...
 import tracker from 'app/utils/time_tracker';
+import {autoUpdateTimezone} from 'app/utils/timezone';
 import {stripTrailingSlashes} from 'app/utils/url';
 
 import LocalConfig from 'assets/config';
 
 const {StatusBarManager} = NativeModules;
 const AUTHENTICATION_TIMEOUT = 5 * 60 * 1000;
 
+export const mattermostStore = configureStore(initialState);
```
[the remark this answers is not in the document]
Comment (miguelespinoza) [reply app/mattermost.js]: Yeh it was necessary to reuse the store in other parts of the app like utils/timezone.js

File: app/mattermost.js [hunk 4]
```diff
@@ -399,6 +403,11 @@ export default class Mattermost {
                 Client4.setUserId(currentUserId);
             }
 
+            const currentUser = getCurrentUser(getState());
```
[the remark this answers is not in the document]
Comment (miguelespinoza) [reply app/mattermost.js]: Nice catch 👍

File: app/screens/timezone/select_timezone/select_timezone.js (hunk 1 above)
Comment (miguelespinoza) [reply app/screens/timezone/select_timezone/select_timezone.js]: Yeh looks like that's an issue with the dev branch.  Removing it did the trick

File: app/screens/timezone/timezone.js (hunk 2 above)
Comment (miguelespinoza) [reply app/screens/timezone/timezone.js]: 🤦‍♂️

**Pull request closed without merging**